### ISSUE TITLE
Check server EOF deprecated capability

### ISSUE THREAD

**Issue by mp911de** [mirromutth/r2dbc-mysql#26]
I tried to find out which version are supported. Using 5.6 fails for me, using 5.7.22 works.

Trying to log into version 5.6.43 fails with:

```
2019-07-05 14:48:54,434 ERROR th.r2dbc.mysql.client.ReactorNettyClient: 113 - Connection Error: unknown message header 254 on command phase
io.r2dbc.spi.R2dbcNonTransientResourceException: unknown message header 254 on command phase
	at io.github.mirromutth.r2dbc.mysql.message.server.ServerMessageDecoder.decodeCommandMessage(ServerMessageDecoder.java:200)
	at io.github.mirromutth.r2dbc.mysql.message.server.ServerMessageDecoder.decodeMessage(ServerMessageDecoder.java:83)
	at io.github.mirromutth.r2dbc.mysql.message.server.ServerMessageDecoder.decode(ServerMessageDecoder.java:55)
	at io.github.mirromutth.r2dbc.mysql.client.MessageDuplexCodec.channelRead(MessageDuplexCodec.java:83)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:374)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:360)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:352)
	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:323)
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:297)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:374)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:360)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:352)
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1408)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:374)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:360)
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:930)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:163)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:682)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:617)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:534)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:496)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:906)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at java.lang.Thread.run(Thread.java:748)
```

**Comment by mirromutth** [comment]
Looks like MySQL server 5.6.x sent `EofMessage` in command phase whatever client mark EOF message to deprecated.

Would this unit test been querying `INSERT / UPDATE / DELETE`? (or `SET @@some_variable`, table definition)

**Comment by mp911de** [comment]
Take a look at the tests:

* [`AbstractDatabaseClientIntegrationTests`](https://github.com/spring-projects/spring-data-r2dbc/blob/master/src/test/java/org/springframework/data/r2dbc/core/AbstractDatabaseClientIntegrationTests.java)
* [`AbstractTransactionalDatabaseClientIntegrationTests`](https://github.com/spring-projects/spring-data-r2dbc/blob/master/src/test/java/org/springframework/data/r2dbc/core/AbstractTransactionalDatabaseClientIntegrationTests.java)
* [`AbstractR2dbcRepositoryIntegrationTests`](https://github.com/spring-projects/spring-data-r2dbc/blob/master/src/test/java/org/springframework/data/r2dbc/repository/AbstractR2dbcRepositoryIntegrationTests.java)

**Comment by mirromutth** [comment]
MySQL 5.6.x has no deprecate EOF flag, we cannot deprecate EOF in MySQL 5.6.x.

This problem is likely to be magnified because all current logic implementations (like `MySqlResult`, `returnGeneratedValues`, etc.) are based on OK message instead of EOF.

**Comment by mirromutth** [comment]
Note: It does not affect `LAST_INSERT_ID` because `INSERT/UPDATE/DELETE` still uses OK message as the end signal.

**Comment by mirromutth** [comment]
Execute me, could you try to run a unit test again? PR has merged, I have ran my unit tests and it worked for now.

**Comment by mirromutth** [comment]
@mp911de According to my plan, I will build next milestone this week, if any newer problems happen which about EOF message, please comment on this issue. This issue will be keep.

**Comment by mirromutth** [comment]
This problem has been resolved by #32 and due to inactivity, so I try to close this.
